### PR TITLE
Add browser validation

### PR DESCRIPTION
**Pull request by dianakrog**

### CODE REVIEW

File: src/test/resources/Features/searchGoogle.feature [hunk 1]
```diff
@@ -3,8 +3,13 @@
 Feature: Search in Google
 
   Scenario Outline: Valid Search
-    Given browser is open 
+    Given "<browser>" is open
```
Comment (aperdomob) [inline src/test/resources/Features/searchGoogle.feature]: the browser should not be part of the gherkin definition

File: src/test/java/stepDefinitions/searchSteps.java [hunk 2]
```diff
@@ -2,35 +2,50 @@
...
+			driver = wdmC.capabilities(optionsC).create();
+			break;
+
+		case "firefox":
+
+			FirefoxOptions optionsF = new FirefoxOptions();
+			driver = wdmF.capabilities(optionsF).create();
+			break;
+
+		}
```
Comment (aperdomob) [inline src/test/java/stepDefinitions/searchSteps.java]: move these logic to another class

File: src/test/java/stepDefinitions/searchSteps.java [hunk 3]
```diff
@@ -2,35 +2,50 @@
...
+import io.github.bonigarcia.wdm.WebDriverManager;
 
 public class searchSteps {
 
 	WebDriver driver = null;
 
-	@Given("browser is open")
-	public void browser_is_open() {
+		WebDriverManager wdmC = WebDriverManager.chromedriver();
+		WebDriverManager wdmF = WebDriverManager.firefoxdriver();
```
Comment (aperdomob) [inline src/test/java/stepDefinitions/searchSteps.java]: avoid create two drivers and only uses one

**Review by aperdomob — CHANGES_REQUESTED** [review]

**Comment by aperdomob** [comment]
an additional comment is to remove geckodriver.exe from your project

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -13,24 +13,27 @@ jobs:
...
-        path: './target/HtmlReports.html'
+        path: './target/generatedreport'
     - name: Display structure of downloaded files
       run: ls -R
       working-directory: './target'
     - uses: actions/upload-artifact@v2
       with:
-        name: my-reportTest
-        path: './target/HtmlReports.html'
+        name: 'my-reportTest${{matrix.browser}}'
```
Comment (aperdomob) [inline .github/workflows/CI.yml:38]: I don't understand these changes

Comment (aperdomob) [inline .github/workflows/CI.yml:38]: I don't understand these changes

File: src/test/java/Utils/TypeBrowser.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package Utils;
+
+public class TypeBrowser {
+	
+	protected static final String browser = System.getProperty("exeBrowser","chrome");
+	
+}
```
Comment (aperdomob) [inline src/test/java/Utils/TypeBrowser.java]: why do you need this class

File: src/test/java/Utils/getDriver.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxOptions;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class getDriver {
+
+	public static WebDriver driver;
+
+	public static void iniciarConfiguracion() throws Exception{
```
Comment (aperdomob) [inline src/test/java/Utils/getDriver.java]: use english

Comment (aperdomob) [reply src/test/java/Utils/getDriver.java]: pass the browser name like paramerts

File: src/test/java/Utils/getDriver.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+			
+			break;
+		case "firefox":
+			WebDriverManager wdmF = WebDriverManager.firefoxdriver();
+			FirefoxOptions optionsF = new FirefoxOptions();
+			driver = wdmF.capabilities(optionsF).create();
+			driver.manage().window().maximize();
+			break;
+		default:
+			throw new Exception("Navegador " + TypeBrowser.browser + " No soportado" );
```
Comment (aperdomob) [inline src/test/java/Utils/getDriver.java]: use english

File: src/test/java/stepDefinitions/searchSteps.java [hunk 8]
```diff
@@ -2,62 +2,55 @@
```
Comment (aperdomob) [inline src/test/java/stepDefinitions/searchSteps.java:2]: why this class has a lot of modifications?

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: src/test/java/Utils/TypeBrowser.java (hunk 5 above)
Comment (dianakrog) [reply src/test/java/Utils/TypeBrowser.java]: Clase removida. Se ajusta clase getDriver para soportar lla tarea de TypeBrowser

File: src/test/java/Utils/getDriver.java (hunk 6 above)
Comment (dianakrog) [reply src/test/java/Utils/getDriver.java]: Done

File: src/test/java/Utils/getDriver.java (hunk 7 above)
Comment (dianakrog) [reply src/test/java/Utils/getDriver.java]: Done

File: src/test/java/stepDefinitions/searchSteps.java (hunk 8 above)
Comment (dianakrog) [reply src/test/java/stepDefinitions/searchSteps.java:2]: Alternative solutions were tested to pass the browser as a parameter. Were discarded

File: .github/workflows/CI.yml (hunk 4 above)
Comment (dianakrog) [reply .github/workflows/CI.yml:38]: I change the report, it is generated in a folder. Variable is sent to differentiate the reports

Comment (dianakrog) [reply .github/workflows/CI.yml:38]: I change the report, it is generated in a folder. Variable is sent to differentiate the reports

File: src/test/java/Utils/getDriver.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxOptions;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class getDriver {
+
+	public static WebDriver driver;
```
Comment (aperdomob) [inline src/test/java/Utils/getDriver.java]: use encapsulation

File: src/test/java/Utils/getDriver.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
+package Utils;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxOptions;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class getDriver {
```
Comment (aperdomob) [inline src/test/java/Utils/getDriver.java]: `getDriver` does not have the naming conventions to Java Class

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: src/test/java/Utils/BrowserManager.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class BrowserManager {
+
+	public static WebDriver startConfig(){
+
+		switch (System.getProperty("exeBrowser","chrome")) {
+		case "chrome":
+			WebDriverManager wdmC = WebDriverManager.chromedriver();
```
Comment (aperdomob) [inline src/test/java/Utils/BrowserManager.java]: what is C, use complete names. Avoid subjectivity

File: src/test/java/Utils/BrowserManager.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+			ChromeOptions optionsC = new ChromeOptions();
+			WebDriver driverChrome = wdmC.capabilities(optionsC).create();
+			return driverChrome;
```
Comment (aperdomob) [inline src/test/java/Utils/BrowserManager.java]: you can resume these lines with
```java
return  WebDriverManager.chromedriver().capabilities(new ChromeOptions()).create();
```

File: src/test/java/Utils/BrowserManager.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+		case "firefox":
+			WebDriverManager wdmF = WebDriverManager.firefoxdriver();
+			FirefoxOptions optionsF = new FirefoxOptions();
+			WebDriver driverFirefox = wdmF.capabilities(optionsF).create();
+			return driverFirefox;
```
Comment (aperdomob) [inline src/test/java/Utils/BrowserManager.java]: same comment here

**Review by aperdomob — CHANGES_REQUESTED** [review]
Also remove geckodriver.exe

File: src/test/java/Utils/BrowserManager.java (hunk 11 above)
Comment (dianakrog) [reply src/test/java/Utils/BrowserManager.java]: summary expression in return

**Review by aperdomob — APPROVED** [review]

**Pull request merged**